lib/ContractInteract/BrandedToken.js

```diff
@@ -0,0 +1,393 @@
+'use strict';
+
+const Web3 = require('web3');
+const Contracts = require('../Contracts');
+const AbiBinProvider = require('../AbiBinProvider');
+const Utils = require('../../utils/Utils');
+
+const ContractName = 'BrandedToken';
+
+/**
+ * Contract interact for Branded token.
+ */
+class BrandedToken {
+  /**
+   * Constructor for Branded token.
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {string} address BrandedToken contract address.
+   */
+  constructor(web3, address) {
+    if (!(web3 instanceof Web3)) {
+      throw new TypeError("Mandatory Parameter 'web3' is missing or invalid");
+    }
+    if (!Web3.utils.isAddress(address)) {
+      throw new TypeError(
+        `Mandatory Parameter 'address' is missing or invalid: ${address}`,
+      );
+    }
+
+    this.web3 = web3;
+    this.address = address;
+
+    this.contract = Contracts.getBrandedToken(this.web3, this.address);
+
+    if (!this.contract) {
+      throw new TypeError(
+        `Could not load utility branded token contract for: ${this.address}`,
+      );
+    }
+
+    this.convertToBrandedTokens = this.convertToBrandedTokens.bind(this);
+    this.requestStake = this.requestStake.bind(this);
+    this.requestStakeRawTx = this.requestStakeRawTx.bind(this);
+    this.acceptStakeRequest = this.acceptStakeRequest.bind(this);
+    this.acceptStakeRequestRawTx = this.acceptStakeRequestRawTx.bind(this);
+    this.liftRestriction = this.liftRestriction.bind(this);
+    this.liftRestrictionRawTx = this.liftRestrictionRawTx.bind(this);
+    this.isUnrestricted = this.isUnrestricted.bind(this);
+    this.rejectStakeRequest = this.rejectStakeRequest.bind(this);
+    this.rejectStakeRequestRawTx = this.rejectStakeRequestRawTx.bind(this);
```

Comment:
missing (from https://github.com/OpenSTFoundation/brandedtoken-contracts/blob/develop/contracts/BrandedToken.sol):
* `liftAllRestrictions`
* `revokeStakeRequest`
* `redeem`
* `setSymbol`
* `setName`
* `transfer`
* `transferFrom`
* `convertToValueTokens`

Reply:
Intentionally skipped, as they are not used currently and of low priority for now.

Reply:
Hm... Is there another ticket for them? How do we communicate the lack of methods to the consumer?

Reply:
Refer #104 
Note: There is a default way to access all the contract methods by `this.contract.methods.xyz`
Note should be added in the readme that for a missing method of contract interact use `this.contract.methods.xyz`

Reply:
Is there another ticket for the getters of public variables? They are also missing from other ContractInteracts.